activestorage/lib/active_storage/service/registry.rb

```diff
@@ -7,7 +7,8 @@ def initialize(configurations)
       @services = {}
     end
 
-    def fetch(name)
+    def fetch(service)
+      name = service.is_a?(Proc) ? service.call : service
```

Comment:
I'm not passing any arguments for my usecase. Do we have `record` magically available here? We probably should allow passing an argument to `service.call` here.

Reply:
given the original implementation, it seems we *must* pass the record to the services proc, so just adding it in here as well.